rust/hyperlane-core/src/traits/mod.rs

```diff
@@ -60,3 +62,18 @@ impl From<ethers_core::types::TransactionReceipt> for TxOutcome {
         }
     }
 }
+
+impl TryFrom<cosmrs::proto::cosmos::base::abci::v1beta1::TxResponse> for TxOutcome {
+    type Error = ChainCommunicationError;
+
+    fn try_from(
+        response: cosmrs::proto::cosmos::base::abci::v1beta1::TxResponse,
+    ) -> ChainResult<Self> {
+        Ok(Self {
+            transaction_id: H256::from_slice(hex::decode(response.txhash)?.as_slice()).into(),
+            executed: response.code == 0,
+            gas_used: U256::from(response.gas_used),
+            gas_price: U256::from(response.gas_wanted),
```

Comment:
I know this wasn't your change but the gas wanted is the gas limit for tx and not the gas price

I assume this was originally set like this because it wasn't available in `TxResponse` - imo it's fine to just set it to 1 or something for now (or to get it from the ceil of here https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/trevor/new-featv3-cosmos-oct-28/rust/chains/hyperlane-cosmos/src/providers/grpc.rs#L36, which is how we set the gas price atm - unfortunately there doesn't seem to be a reasonable way of getting it via RPC so atm it's just hardcoded :/), but to add a comment about it

Reply:
I see :/ `DEFAULT_GAS_PRICE` isn't available here because it lives in hyperlane-cosmos, and the TryFrom implementation has to live in this crate due to the orphan rule. I'll just set it to `U256::one()`